types/json2csv/json2csv-tests.ts

```diff
@@ -36,11 +42,62 @@ const opts: json2csv.Options<Car> = {
     }]
 };
 
-const json2csvParser = new Parser<Car>(opts);
+const data = [{
+    car: 'VW Beetle',
+    price: 1395
+}, {
+    car: 'VW T1',
+    price: 1357
+}];
 
-parser.parse([]);
-const transform: Transform<ExampleObj> = new Transform<ExampleObj>({ quote: '' });
-const nodeTransform: NodeTransform = transform;
+// Test for Synchronous Parser
+try {
+    const parser = new Parser(opts);
+    const csv = parser.parse({ car: '', price: 1 });
+    console.log(csv);
+} catch (err) {
+    console.error(err);
+}
+
+// Test convenience method "parse"
+try {
+    const csv = parse(data, opts);
+    console.log(csv);
+} catch (err) {
+    console.error(err);
+}
+
+// Test for Asynchronous Parser
+const transformOpts = { highWaterMark: 8192 };
+
+const asyncParser = new AsyncParser(opts, transformOpts);
+
+let csv = '';
+asyncParser.processor
+    .on('data', chunk => (csv += chunk.toString()))
+    .on('end', () => console.log(csv))
+    .on('error', err => console.error(err));
+
+// Test for transform events
+asyncParser.transform
+    .on('header', header => console.log(header))
+    .on('line', line => console.log(line))
+    .on('error', err => console.log(err));
+
+asyncParser.input.push(data); // This data might come from an HTTP request, etc.
+asyncParser.input.push(null); // Sending `null` to a stream signal that no more data is expected and ends it.
+
+const input = createReadStream('/path/to/input', { encoding: 'utf8' });
+const output = createWriteStream('path/to/output', { encoding: 'utf8' });
+
+asyncParser.fromInput(input).toOutput(output).promise()
+    .then(csv => console.log(csv))
+    .catch(err => console.error(err));
+
+// Test convenience method "parseAsync"
+parseAsync(data, opts)
```

Comment:
There should also be a test where data is an object and where data is a readable stream